test/find.test.ts

```diff
@@ -1,11 +1,26 @@
-import * as path from 'path';
+import { Stats } from 'fs';
 
-import { find } from '../lib/find';
+import * as path from 'path';
+import * as findModule from '../lib/find';
 
 describe('find', () => {
+  it('find should throw if dirStat is not a directory', async () => {
+    const statSpy = jest.spyOn(findModule, 'stat');
+    const statIsNotDirectory = {
+      isFile: () => true,
+      isDirectory: () => false,
+      size: 0,
+    } as Stats;
+
+    statSpy.mockResolvedValueOnce(statIsNotDirectory);
+    await expect(findModule.find('workspace')).rejects.toThrow(
+      'workspace is not a directory',
+    );
+  });
```

Comment:
(this is a good to have to validate that find is throwing properly)

Reply:
did we lack that test since before?

Reply:
Yep we missed it